### ISSUE TITLE
Add new <generatedAnnotationType/> flag to specify Generated annotation

### ISSUE THREAD

**Issue by dustContributor** [jOOQ/jOOQ#7581]
### Expected behavior and actual behavior:

At least on previous versions of codegen (3.10.x), the Generated annotation was imported like
```
import javax.annotation.Generated;
```
Which worked fine on Java 8 as far as I've tried. jOOQ 3.11.0 codegen generates the same imports, but if you're using Java 10, it seems the annotation should be imported like:
```
import javax.annotation.processing.Generated;
```
Otherwise it cant find it. I'm not sure if it was moved between releases.
### Steps to reproduce the problem (include example code if possible):
Use codegen to generate any classes since they all use the Generated annotation.
### Versions:

- jOOQ: 3.11.0
- Java: 10
- Database (include vendor): MySQL 5.6.21
- OS: Debian x86_64
- JDBC Driver (include name if inofficial driver): MySQL Connector 5.1.43

**Comment by lukaseder** [comment]
Thanks for your report. The `javax.annotation.Generated` annotation (since Java 6) is in the `java.xml.ws.annotation` module, whereas the new `javax.annotation.processing.Generated` annotation (since Java 9) is in the `java.compiler` module.

I don't see any particular reason why one should be preferred over the other apart from the fact that the former is slightly more powerful than the latter:

```java
// javax.annotation.Generated
@Target({PACKAGE, TYPE, ANNOTATION_TYPE, METHOD, CONSTRUCTOR, FIELD, LOCAL_VARIABLE, PARAMETER})

// javax.annotation.processing.Generated
@Target({PACKAGE, TYPE,                  METHOD, CONSTRUCTOR, FIELD, LOCAL_VARIABLE, PARAMETER})
```

... whereas the latter is located in a slightly more generally useful module than the former.

The fact that this simple annotation requires a separate module from Java 9 onwards is a good reason not to use the annotation at all, or perhaps, to replicate it as a jOOQ annotation `org.jooq.Generated`.

**Comment by lukaseder** [comment]
Rolling our own annotation might not be the best idea given that IDEs might pick up the annotation and process it, e.g. to hint at users that they shouldn't manually modify generated sources...

**Comment by lukaseder** [comment]
Hmm, the `java.xml.ws.annotation` module is deprecated for removal. Perhaps we should move to the `javax.annotation.processing.Generated` one, indeed...

**Comment by lukaseder** [comment]
This will be fixed through #7583
